### PR TITLE
Autocomplete creatable: improved API to cover more use cases

### PR DESCRIPTION
**Pull request by vmilan**
# Description

Shortcut: https://app.shortcut.com/cartoteam/story/418320/design-system-autocomplete-creatable-support-all-menuitem-options
[sc-418320]

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,11 +2,14 @@
 
 ## Not released
 
+- Forms fixes & improvements [#881](https://github.com/CartoDB/carto-react/pull/881)
+- Fix LegendProportion radius scale [#877](https://github.com/CartoDB/carto-react/pull/877)
```
Comment (vmilan) [inline CHANGELOG.md:6]: Missing entry from previous PR

File: packages/react-ui/src/components/molecules/Autocomplete.d.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ export type AutocompleteProps<
 > = MuiAutocompleteProps<Value, Multiple, DisableClearable, FreeSolo, ChipComponent> & {
   creatable?: boolean;
   newItemTitle?: React.ReactNode | string;
```
Comment (Josmorsot) [inline packages/react-ui/src/components/molecules/Autocomplete.d.ts:13]:
```suggestion
  newItemTitle?: string | (value: string) => (React.ReactNode | string);
```

File: packages/react-ui/src/components/molecules/Autocomplete.js [hunk 3]
```diff
@@ -42,9 +44,7 @@ const Autocomplete = forwardRef(
       if (inputValue.length > 1 && inputValue !== '' && !isExisting) {
         filtered.push({
           inputValue,
-          title:
-            newItemTitle ||
-            `${intlConfig.formatMessage({ id: 'c4r.form.add' })} "${inputValue}"`
+          title: `${intlConfig.formatMessage({ id: newItemTitle })} "${inputValue}"`
```
Comment (Josmorsot) [inline packages/react-ui/src/components/molecules/Autocomplete.js:47]:
```suggestion
          title: typeof newItemTitle === 'function' ? newItemTitle(inputValue) : `${newItemTitle} "${inputValue}"`
```

**Review by VictorVelarde — APPROVED** [review]
👍🏻

**Review by AlvaroAlmonacidRojo — APPROVED** [review]

**Review by Josmorsot — APPROVED** [review]

**Pull request merged**